app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntryEffectHandler.kt

```diff
@@ -37,110 +38,84 @@ import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator.Result
 import org.simple.clinic.widgets.ageanddateofbirth.UserInputDateValidator.Result.Valid
 import org.threeten.bp.Instant
 import org.threeten.bp.LocalDate
-
-object BloodPressureEntryEffectHandler {
+import java.util.UUID
+
+class BloodPressureEntryEffectHandler(
+    private val ui: BloodPressureEntryUi,
+    private val userSession: UserSession,
+    private val facilityRepository: FacilityRepository,
+    private val patientRepository: PatientRepository,
+    private val bloodPressureRepository: BloodPressureRepository,
+    private val appointmentsRepository: AppointmentRepository,
+    private val userClock: UserClock,
+    private val paddingCharacter: UserInputDatePaddingCharacter,
+    private val schedulersProvider: SchedulersProvider
+) {
   private val reportAnalyticsEvents = ReportAnalyticsEvents()
 
-  fun create(
-      ui: BloodPressureEntryUi,
-      userSession: UserSession,
-      facilityRepository: FacilityRepository,
-      patientRepository: PatientRepository,
-      bloodPressureRepository: BloodPressureRepository,
-      appointmentsRepository: AppointmentRepository,
-      userClock: UserClock,
-      inputDatePaddingCharacter: UserInputDatePaddingCharacter,
-      schedulersProvider: SchedulersProvider
-  ): ObservableTransformer<BloodPressureEntryEffect, BloodPressureEntryEvent> {
-    val createNewBpEntryTransformer = createNewBpEntryTransformer(
-        userSession,
-        facilityRepository,
-        patientRepository,
-        bloodPressureRepository,
-        appointmentsRepository,
-        userClock,
-        schedulersProvider.io()
-    )
-
-    val updateBpEntryTransformer = updateBpEntryTransformer(
-        userSession,
-        facilityRepository,
-        patientRepository,
-        bloodPressureRepository,
-        userClock
-    )
-
+  fun create(): ObservableTransformer<BloodPressureEntryEffect, BloodPressureEntryEvent> {
```

Comment:
Discussed offline: Make this a factory function instead of an instance method.

Reply:
This is done.